packages/core/services/DatabaseService/index.ts

```diff
@@ -92,6 +94,70 @@ export default abstract class DatabaseService {
         this.dataSourceToAnnotationsMap.clear();
     }
 
+    protected async checkDataSourceForErrors(name: string): Promise<string[]> {
+        const tableColumnsQueryResult = await this.query(`\
+            SELECT COLUMN_NAME                \
+            FROM INFORMATION_SCHEMA.COLUMNS   \
+            WHERE TABLE_NAME = '${name}'      \
+        `);
+        const columnsOnTable = tableColumnsQueryResult.reduce(
+            (accum: Set<string>, row) => accum.add(row.column_name),
+            new Set<string>()
+        );
+
+        if (!columnsOnTable.has("File Path")) {
+            let error =
+                '\
+                "File Path" column is missing in the data source. \
+                Check the data source header row for a "File Path" (case-sensitive) column name and try again.';
+
+            // Attempt to find a column with a similar name to "File Path"
+            const columns = Array.from(columnsOnTable);
+            const filePathLikeColumn =
+                columns.find((column) => column.toLowerCase().includes("path")) ||
+                columns.find((column) => column.toLowerCase().includes("file"));
+            if (filePathLikeColumn) {
+                error += ` Found a column with a similar name: "${filePathLikeColumn}".`;
+            }
+
+            // Unable to determine if the file path is empty or not
+            // when it is not present so return here before checking
+            // for other errors
+            return [error];
+        }
+
+        const errors: string[] = [];
+        const isFilePathEverBlankQueryResult = await this.query(`\
+            SELECT EXISTS (                      \
+                FROM "${name}"                   \
+                WHERE TRIM("File Path") IS NULL  \
+            ) AS result                          \
+        `);
+        if (isFilePathEverBlankQueryResult[0]["result"]) {
+            errors.push('"File Path" column contains empty or purely whitespace values.');
```

Comment:
Could we upgrade this to count the errors, or ideally list a value from the problem rows?  If my source has a million rows, I don't want to find the errors by hand.  Does the client have access to the SQL?  I think the sql is a function of the app using a library to be smart about csv.

Reply:
Yeah we could probably list the rows, good idea